Glm intro pr (#51)
courtiol/LM2GLMM: vignettes/GLM_intro_course.Rmd

### You can predict outcomes for 2 different scales

> - for the scale of the **linear predictor** (default for glm), e.g.:

Reviewer: ```suggestion
- for the scale of the **linear predictor** (default for `glm()`), e.g.:
```

Author: 👍 I notice that in most cases we're also specifying the NAMESPACE (i.e. `stats::glm()` and `spaMM::fitme()`), so I'll also do that here for consistency.

pred <- predict(fit_poiss, newdata = newdata, se.fit = TRUE)
lwr_eta <- pred$fit + qnorm(0.025) * pred$se.fit
upr_eta <- pred$fit + qnorm(0.975) * pred$se.fit
cbind(pred = poisson(link = "log")$linkinv(pred$fit),
      lwr = poisson(link = "log")$linkinv(lwr_eta),
      upr = poisson(link = "log")$linkinv(upr_eta))
```
Notes:

- don't use `type = "response"` in `predict()` because transformations need to happen after CI are computed!

- for proper predictions involving more predictors, better use `pdep_effects()` on a model fitted with **{spaMM}** `r .emo("recap")`

## CI using Gaussian approximation `r .emo("practice")`

CIs are assumed symmetrical (i.e. Gaussian) on the **linear predictor scale** but not on the **response scale**.

```{r conf fitted poiss response}
newdata <- data.frame(humans_eaten = 0:1)
pred_wrong <- predict(fit_poiss, newdata = newdata, se.fit = TRUE, type = "response")

> lwr_eta <- pred_wrong$fit + qnorm(0.025) * pred_wrong$se.fit

Reviewer: ```suggestion
lwr_eta <- pred_wrong$fit + qnorm(0.025) * pred_wrong$se.fit ## Wrong: for example purposes only
```

> upr_eta <- pred_wrong$fit + qnorm(0.975) * pred_wrong$se.fit

Reviewer: ```suggestion
upr_eta <- pred_wrong$fit + qnorm(0.975) * pred_wrong$se.fit ## Wrong: for example purposes only
```

                        upr = poisson(link = "log")$linkinv(upr_eta),
                        method = "link")
#Wrong method. CIs on response scale
pred_wrong <- predict(fit_poiss, newdata = newdata, se.fit = TRUE, type = "response")
lwr_eta <- pred_wrong$fit + qnorm(0.025) * pred_wrong$se.fit
upr_eta <- pred_wrong$fit + qnorm(0.975) * pred_wrong$se.fit
plot_data <- rbind(plot_data,
                   data.frame(humans_eaten = 0:1,
                              eggs = pred_wrong$fit,
                              lwr = lwr_eta,
                              upr = upr_eta,
                              method = "response"))

ggplot(data = plot_data) +
  geom_hline(yintercept = 0) +
  geom_errorbar(aes(x = humans_eaten, ymin = lwr, ymax = upr, group = method),
                width = 0.25, position = position_dodge(width = 0.25)) +
  geom_point(aes(x = humans_eaten, y = eggs, group = method, shape = method),
             size = 3, fill = "grey",
             position = position_dodge(width = 0.25)) +
  scale_shape_discrete(labels = c("Linear predictor scale", "Response scale"), name = "") +
  scale_x_continuous(breaks = c(0, 1)) +
  theme_classic()

> ```

Reviewer: I think that showing what not to do is confusing, because they will forget it is the wrong thing and do it. Don't you think?

Author: Perhaps, but we already do this many times in the course right? (e.g. predict on mean values). IMO, I find this particular example really useful because it shows that using response is actively *wrong* (i.e. we can get impossible values) and also it re-emphasises another difference between link and response scale (i.e. that CIs are symmetrical on link but not response).

However, I'm happy to leave this to your discretion. If you don't think it helps feel free to remove 😄

Reviewer: Let's keep it for now, with all the bells and whistles.

* there are specific solutions if the origin is zero-augmentation

> ## Modeling simple overdispersion

Reviewer: I meant simply as in a simple way of modelling overdispersion.

Author: Ahh yea I see. 'Modeling simply overdispersion' is not grammatically correct, but I think 'Simple modeling of overdispersion' (which is actually used in the next slide!) would be good.

Reviewer: do that

### What is it?

> It occurs for any event when there are too many zeros.

Reviewer: ```suggestion
It occurs for when there are more zeros than expected under the probability distribution of the data generation process.
```

Fit an **hurdle model**

* e.g. binomial + truncated Poisson or truncated negative binomial

> * a single source of zeros (modelled by binomial)

Reviewer: ```suggestion
* a single source of zeros (usually modelled by a binomial)
```

* e.g. of response variable: number of offspring

<br>

Fit a **zero-inflation model**

* e.g. binomial + Poisson or negative binomial

> * two sources of zeros (modelled by binomial AND Poisson/negative binomial)

Reviewer: ```suggestion
* two sources of zeros (usually modelled by a binomial AND a Poisson or a negative binomial)
```
